### PR TITLE
[AC-5484] Add home program bin

### PR DESCRIPTION
**Pull request by jonkiparsky**
Probably some improvements to make, both in the new code and in the base code. Maybe do those pairing?

### CODE REVIEW

File: allocate.py [hunk 1]
```diff
@@ -54,9 +56,12 @@ def work_left(bins):
 industry_bins = [IndustryBin(industry=value[0],
                              value=4*BIN_DEFAULT)
                  for value in industry.values]
+program_bins = [HomeProgramBin(program=value,
+                               value=3*BIN_DEFAULT)
+                for value, _ in program.values]
```
Comment (mc-nathan) [inline allocate.py]: I'd make these parallel.  I like yours slightly better so let's do that.

File: classes/entity.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
                "commitment",
                "completed"]
 CSV_HEADER = ",".join(CSV_COLUMNS)
-CSV_FORMAT = "{%s}" % "},{".join(CSV_COLUMNS) + "}"
+CSV_FORMAT = "{%s}" % ("},{".join(CSV_COLUMNS))
```
Comment (mc-nathan) [inline classes/entity.py:12]: Good catch.

File: classes/home_program_bin.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        self.program = program
+
+    def __str__(self):
+        return self.name_format.format(self.program)
+
+    def match(self, startup):
+        return startup.properties['program'] == self.program
+
+    def value(self, judge):
+        if judge.properties['industry'] == self.program:
```
Comment (mc-nathan) [inline classes/home_program_bin.py]: s/industry/program

File: classes/home_program_bin.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            return super().value(judge)
+        else:
+            #None? mixed-type returns are ugly
```
Comment (mc-nathan) [inline classes/home_program_bin.py]: Could do `return BIN_NO_VALUE` from the bin.py.

File: classes/judge.py [hunk 5]
```diff
@@ -17,7 +17,7 @@ def __init__(self, data=None):
         self.type = "judge"
         for property in Property.all_properties:
             self.add_property(property, data)
-        self.remaining = int(self.properties.get("commitment", 50))
+        self.remaining = int(self.properties.get("commitment", 50) or 0)
```
Comment (mc-nathan) [inline classes/judge.py]: What is the `or 0` doing?

**Review by mc-nathan — CHANGES_REQUESTED** [review]

File: classes/home_program_bin.py (hunk 3 above)
Comment (jonkiparsky) [reply classes/home_program_bin.py]: doh! caught in copy-paste!

File: classes/judge.py (hunk 5 above)
Comment (jonkiparsky) [reply classes/judge.py]: I had a failure where it tried to make "" into an int. This is a quick way to hack around that.

Comment (mc-nathan) [reply classes/judge.py]: Looks like the generator script is broken.  I'll get that fixed.

**Review by mc-nathan — APPROVED** [review]

**Pull request merged**